src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationILGen.cs

```diff
@@ -248,12 +249,12 @@ internal void GenerateSupportedTypes(Type[] types, TypeBuilder serializerContrac
                     continue;
                 if (!type.GetTypeInfo().IsPublic && !type.GetTypeInfo().IsNestedPublic)
                     continue;
-                if (uniqueTypes[type] != null)
+                if (uniqueTypes.Contains(type))
                     continue;
                 // DDB172141: Wrong generated CS for serializer of List<string> type
                 if (type.GetTypeInfo().IsGenericType || type.GetTypeInfo().ContainsGenericParameters)
                     continue;
-                uniqueTypes[type] = type;
+                uniqueTypes.Add(type);
```

Comment:
Nit: Same comment about using single call to `HashSet<T>.Add` instead of `Contains` & `Add`.